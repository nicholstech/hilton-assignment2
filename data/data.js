const data = {
  rooms: [
    {
      id: 1,
      children: [
        { value: '0', displayValue: '0' },
        { value: '1', displayValue: '1' },
        { value: '2', displayValue: '2' }
      ],
      adults: [
        { value: '1', displayValue: '1' },
        { value: '2', displayValue: '2' }
      ],
      selectedChildren: 0,
      selectedAdults: 0,
      isSelected: false
    },
    {
      id: 2,
      children: [
        { value: '0', displayValue: '0' },
        { value: '1', displayValue: '1' },
        { value: '2', displayValue: '2' }
      ],
      adults: [
        { value: '1', displayValue: '1' },
        { value: '2', displayValue: '2' }
      ],
      selectedChildren: 0,
      selectedAdults: 0,
      isSelected: false
    },
    {
      id: 3,
      children: [
        { value: '0', displayValue: '0' },
        { value: '1', displayValue: '1' },
        { value: '2', displayValue: '2' }
      ],
      adults: [
        { value: '1', displayValue: '1' },
        { value: '2', displayValue: '2' }
      ],
      selectedChildren: 0,
      selectedAdults: 0,
      isSelected: false
    },
    {
      id: 4,
      children: [
        { value: '0', displayValue: '0' },
        { value: '1', displayValue: '1' },
        { value: '2', displayValue: '2' }
      ],
      adults: [
        { value: '1', displayValue: '1' },
        { value: '2', displayValue: '2' }
      ],
      selectedChildren: 0,
      selectedAdults: 0,
      isSelected: false
    },
  ]
}
export default data
