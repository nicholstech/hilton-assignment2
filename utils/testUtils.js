
/**
 * @description Return Shallow Wrapper containing nodes(s) with the given data-test value.
 * @param wrapper -Enzyme shallow wrapper to search within.
 * @param value - Value of data-test attribute for search.
 * @returns {ShallowWrapper|ReactWrapper|NodePath|number|never|bigint|T|T}
 */
export const findByTestAttr = (wrapper, value) => {
  return wrapper.find(`[data-test="${value}"]`)
}
