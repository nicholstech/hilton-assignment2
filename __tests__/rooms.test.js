import React from 'react'
import { shallow } from 'enzyme'
import Rooms from '../components/rooms'
import Room from '../components/room'
import data from '../data/data'
import { findByTestAttr } from '../utils/testUtils'

describe('<Rooms/>', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<Room/>)
  })

  it('should not render a <Room/> element when rooms.length === 0', () => {
    const roomComponent = findByTestAttr(wrapper, 'component-room')
    expect(roomComponent).toHaveLength(0)
  })
  it('should render four <Room/> elements', () => {
    const roomComponent = shallow(<Rooms rooms={data.rooms}/>)
    expect(roomComponent).toMatchSnapshot()
  })
})


