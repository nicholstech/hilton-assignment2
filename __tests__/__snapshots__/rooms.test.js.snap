// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rooms/> should render four <Room/> elements 1`] = `
<Fragment>
  <Room
    adults={
      Array [
        Object {
          "displayValue": "1",
          "value": "1",
        },
        Object {
          "displayValue": "2",
          "value": "2",
        },
      ]
    }
    data-test="component-room"
    isSelected={false}
    key="1"
    room={
      Object {
        "adults": Array [
          Object {
            "displayValue": "1",
            "value": "1",
          },
          Object {
            "displayValue": "2",
            "value": "2",
          },
        ],
        "children": Array [
          Object {
            "displayValue": "0",
            "value": "0",
          },
          Object {
            "displayValue": "1",
            "value": "1",
          },
          Object {
            "displayValue": "2",
            "value": "2",
          },
        ],
        "id": 1,
        "isSelected": false,
        "selectedAdults": 0,
        "selectedChildren": 0,
      }
    }
    roomNumber={1}
    selectedAdults={0}
    selectedChildren={0}
  >
    <Component />
    <Component />
    <Component />
  </Room>
  <Room
    adults={
      Array [
        Object {
          "displayValue": "1",
          "value": "1",
        },
        Object {
          "displayValue": "2",
          "value": "2",
        },
      ]
    }
    data-test="component-room"
    isSelected={false}
    key="2"
    room={
      Object {
        "adults": Array [
          Object {
            "displayValue": "1",
            "value": "1",
          },
          Object {
            "displayValue": "2",
            "value": "2",
          },
        ],
        "children": Array [
          Object {
            "displayValue": "0",
            "value": "0",
          },
          Object {
            "displayValue": "1",
            "value": "1",
          },
          Object {
            "displayValue": "2",
            "value": "2",
          },
        ],
        "id": 2,
        "isSelected": false,
        "selectedAdults": 0,
        "selectedChildren": 0,
      }
    }
    roomNumber={2}
    selectedAdults={0}
    selectedChildren={0}
  >
    <Component />
    <Component />
    <Component />
  </Room>
  <Room
    adults={
      Array [
        Object {
          "displayValue": "1",
          "value": "1",
        },
        Object {
          "displayValue": "2",
          "value": "2",
        },
      ]
    }
    data-test="component-room"
    isSelected={false}
    key="3"
    room={
      Object {
        "adults": Array [
          Object {
            "displayValue": "1",
            "value": "1",
          },
          Object {
            "displayValue": "2",
            "value": "2",
          },
        ],
        "children": Array [
          Object {
            "displayValue": "0",
            "value": "0",
          },
          Object {
            "displayValue": "1",
            "value": "1",
          },
          Object {
            "displayValue": "2",
            "value": "2",
          },
        ],
        "id": 3,
        "isSelected": false,
        "selectedAdults": 0,
        "selectedChildren": 0,
      }
    }
    roomNumber={3}
    selectedAdults={0}
    selectedChildren={0}
  >
    <Component />
    <Component />
    <Component />
  </Room>
  <Room
    adults={
      Array [
        Object {
          "displayValue": "1",
          "value": "1",
        },
        Object {
          "displayValue": "2",
          "value": "2",
        },
      ]
    }
    data-test="component-room"
    isSelected={false}
    key="4"
    room={
      Object {
        "adults": Array [
          Object {
            "displayValue": "1",
            "value": "1",
          },
          Object {
            "displayValue": "2",
            "value": "2",
          },
        ],
        "children": Array [
          Object {
            "displayValue": "0",
            "value": "0",
          },
          Object {
            "displayValue": "1",
            "value": "1",
          },
          Object {
            "displayValue": "2",
            "value": "2",
          },
        ],
        "id": 4,
        "isSelected": false,
        "selectedAdults": 0,
        "selectedChildren": 0,
      }
    }
    roomNumber={4}
    selectedAdults={0}
    selectedChildren={0}
  >
    <Component />
    <Component />
    <Component />
  </Room>
</Fragment>
`;
